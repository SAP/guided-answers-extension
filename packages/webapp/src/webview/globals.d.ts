interface Window {
    vscode: {
        postMessage: (message: string | JSON) => void;
    };
}
