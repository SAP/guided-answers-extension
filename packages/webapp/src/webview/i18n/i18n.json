{
    "GUIDED_ANSWER_HEADER": "Guide",
    "SEARCH_GUIDED_ANSWERS": "Search Guided Answers",
    "ALL_ANSWERS": "Home",
    "STEP_BACK": "Step back",
    "RESTART": "Restart",
    "GUIDED_ANSWERS": "Guided Answers",
    "NO_ANSWERS_FOUND": "No answers found",
    "PLEASE_MODIFY_SEARCH": "Please modfiy search",
    "THIS_SOLVED_MY_ISSUE": "This solved my issue",
    "THIS_DID_NOT_SOLVED_MY_ISSUE": "This did not solve my issue",
    "THANK_YOU_TEXT": "We are glad to hear that your issue has been resolved and we hope you enjoy using the Guided Answers application.",
    "THANKS": "Thanks!",
    "HOME": "Home",
    "SEND": "Send",
    "CLOSE": "Close",
    "ISSUE_IS_NOT_RESOLVED": "Issue is not resolved",
    "WE_ARE_SORRY_TO_HEAR_THAT_YOUR_ISSUE_IS_NOT_YET_RESOLVED": "We are sorry to hear that your issue is not yet resolved.",
    "PLEASE_TELL_US_IF_THIS_ANSWER_WAS_HELPFUL": "Please tell us if this answer was helpful",
    "THERE_ARE_SEVERAL_OPTIONS_FOR_GETTING_FURTHER_ASSISTANCE": "There are several options for getting further assistance:",
    "SEARCH_FOR_ANOTHER_GUIDED_ANSWER": "Search for another Guided Answer",
    "START_AN_EXPERT_CHAT": "Start an Expert Chat",
    "SCHUEDULE_AN_EXPERT": "Schedule an Expert",
    "OPEN_AN_INCIDENT": "Open an Incident",
    "ASK_THE_SAP_COMMUNITY": "Ask the SAP Community",
    "FEEDBACK": "Feedback",
    "FEEDBACK_DIALOG_TITLE": "Is this content helpful?",
    "FEEDBACK_DIALOG_SUBTEXT": "If you have suggestions on how to improve this content we would love to hear them!",
    "FEEDBACK_DIALOG_DISCLAIMER": "Your feedback is anonymous, we do not collect any personal data.",
    "FEEDBACK_DIALOG_SUGGESTION": "Your suggestion",
    "FEEDBACK_SENT_DIALOG_TITLE": "Message sent",
    "FEEDBACK_SENT_DIALOG_SUBTEXT": "Thanks for the feedback!"
}
