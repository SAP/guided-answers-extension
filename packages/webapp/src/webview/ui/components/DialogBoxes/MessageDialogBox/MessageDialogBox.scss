.solved-message-dialog {
    .ms-Dialog-main {
        background-color: var(--vscode-editorWidget-background);
    }
    .ms-Dialog-title {
        color: var(--vscode-settings-headerForeground);
        font-weight: 700;
        font-size: 14px;
        line-height: 16px;
        text-align: center;
        margin: 5px 0px 15px 0px;
    }
    .ms-Dialog-subText {
        color: var(--vscode-settings-headerForeground);
        font-style: normal;
        font-weight: 400;
        font-size: 13px;
        line-height: 18px;
        margin-bottom: 0;
        text-align: left;
        margin-top: 0;
    }
    .ms-Dialog-inner {
        padding: 0px 45px;
    }
    .button-container {
        display: flex;
        justify-content: center;
        gap: 10px;
    }

    .default-button {
        background-color: var(--vscode-button-secondaryBackground);
        color: var(--vscode-button-foreground);
        border-color: var(--vscode-button-secondaryBackground);
        text-align: center;
    }

    .primary-button {
        color: var(--vscode-button-foreground);
        font-weight: 400;
        border-color: var(--vscode-button-background);
        background-color: var(--vscode-button-background);
        border-style: none;
    }

    button:focus-visible {
        outline: 1px solid var(--vscode-focusBorder, #007fd4);
        border-style: none;
    }
    .ms-Dialog-actions {
        margin: 25px 0px;
    }
    .ms-Dialog-actionsRight {
        justify-content: center;
    }

    .ms-Dialog-content {
        overflow: hidden;
        padding: 0px;
    }
}
