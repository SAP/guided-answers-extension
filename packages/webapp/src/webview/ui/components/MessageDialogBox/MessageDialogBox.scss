.solved-message-dialog {
    .ms-Dialog-main {
        background-color: var(--vscode-editorWidget-background);
    }
    .ms-Dialog-title {
        color: var(--vscode-settings-headerForeground);
        font-weight: 700;
        font-size: 14px;
        line-height: 16px;
        text-align: center;
    }
    .ms-Dialog-subText {
        color: var(--vscode-settings-headerForeground);
        font-style: normal;
        font-weight: 400;
        font-size: 13px;
        line-height: 18px;
    }
    .actionsRight-122 {
        text-align: center;
    }

    .root-130 {
        background-color: var(--vscode-button-secondaryBackground);
        border-color: var(--vscode-button-secondaryBackground);
    }

    button:focus-visible {
        outline: 1px solid var(--vscode-focusBorder, #007fd4);
    }

    .label-126 {
        color: var(--vscode-button-foreground);
        font-weight: 400;
    }

    button:focus-visible {
        outline: 1px solid var(--vscode-focusBorder, #007fd4);
    }
}
