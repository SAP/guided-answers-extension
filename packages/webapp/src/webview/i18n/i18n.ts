import { use } from 'i18next';
import { initReactI18next } from 'react-i18next';
import * as i18nEn from './i18n.json';

export { i18n } from 'i18next';

export function initI18n(): void {
    use(initReactI18next)
        .init({
            resources: {
                en: {
                    translation: i18nEn
                }
            },
            lng: 'en',
            fallbackLng: 'en',
            interpolation: {
                escapeValue: false
            }
        })
        .catch((error) => console.error(error));
}
