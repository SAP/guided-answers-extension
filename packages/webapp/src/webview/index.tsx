import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { App } from './ui/components/App';
import { actions, store } from './state';
import { initI18n } from './i18n';

/**
 * Initialization of i18n, icons
 */
initI18n();

/**
 * Render app
 */
render(
    <Provider store={store}>
        <App />
    </Provider>,
    document.getElementById('root') as HTMLElement
);

/**
 * Fire action which is captured in hosting app (ide-extension) to signal that we are ready to receive messages
 */
actions.webviewReady();

/**Once webview is ready we focus to search field */
document.getElementById('search-field')?.focus();
