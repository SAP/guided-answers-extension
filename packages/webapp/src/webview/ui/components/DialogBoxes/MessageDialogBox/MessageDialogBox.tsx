import React, { ReactElement, useState, useEffect } from 'react';
import './MessageDialogBox.scss';
import { Dialog, DialogType, DialogFooter } from '@fluentui/react/lib/Dialog';
import { PrimaryButton, DefaultButton } from '@fluentui/react/lib/Button';
import i18next from 'i18next';
import { FocusZone, FocusZoneDirection } from '@fluentui/react-focus';

/**
 * @param props props for Message Dialog Box component
 * @param props.dialogTitle title for dialog box
 * @param props.dialogText text for dialog box
 * @param props.dialogVisible visability status for dialog box
 * @param props.stylingClassName classname for dialog box
 * @param props.primaryButtonAction a function that triggers a redux state change when primary button is pressed
 * @param props.defaultButtonAction a function that triggers a redux state change when default button is pressed
 * @returns - react element to show a Dialog box
 */
export function MessageDialogBox(props: {
    dialogTitle: string;
    dialogText: string;
    stylingClassName: string;
    dialogVisible: boolean;
    primaryButtonAction: Function;
    defaultButtonAction: Function;
}): ReactElement {
    const [isVisible, setVisible] = useState(props.dialogVisible);
    useEffect(() => {
        setVisible(props.dialogVisible);
    }, [props.dialogVisible]);

    const dialogContentProps = {
        type: DialogType.normal,
        title: props.dialogTitle,
        subText: props.dialogText
    };

    const modalProps = {
        isDarkOverlay: true,
        className: `${props.stylingClassName}`
    };

    return (
        <>
            <Dialog hidden={!isVisible} dialogContentProps={dialogContentProps} modalProps={modalProps}>
                <DialogFooter>
                    <FocusZone direction={FocusZoneDirection.horizontal} className="button-container">
                        <PrimaryButton
                            className="primary-button"
                            text={i18next.t('HOME')}
                            onClick={() => props.primaryButtonAction()}
                        />
                        <DefaultButton
                            className="default-button"
                            text={i18next.t('CLOSE')}
                            onClick={() => {
                                props.defaultButtonAction();
                            }}
                        />
                    </FocusZone>
                </DialogFooter>
            </Dialog>
        </>
    );
}
