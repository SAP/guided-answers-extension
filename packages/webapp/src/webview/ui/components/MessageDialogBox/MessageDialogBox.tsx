import React, { ReactElement, useState } from 'react';
import './MessageDialogBox.scss';
import { Dialog, DialogType, DialogFooter } from '@fluentui/react/lib/Dialog';
import { PrimaryButton, DefaultButton } from '@fluentui/react/lib/Button';
import i18next from 'i18next';

/**
 * @param props props for Message Dialog Box component
 * @param props.dialogTitle title for dialog box
 * @param props.dialogText text for dialog box
 * @param props.dialogVisible visability status for dialog box
 * @param props.stylingClassName classname for dialog box
 * @param props.primaryButtonAction a function that triggers a redux state change when primary button is pressed
 * @param props.defaultButtonAction a function that triggers a redux state change when default button is pressed
 * @returns - react element to show a Dialog box
 */
export function MessageDialogBox(props: {
    dialogTitle: string;
    dialogText: string;
    stylingClassName: string;
    dialogVisible: boolean;
    primaryButtonAction: Function;
    defaultButtonAction: Function;
}): ReactElement {
    const [isVisible, setVisible] = useState(props.dialogVisible);

    // const primaryButtonFunc = () => props.primaryButtonAction();
    // const toggleDialog = (): void => {
    // setVisible(!isVisible);
    // if (isVisible === false) {
    //     primaryButtonFunc();
    // }
    // };

    const toggle = (): void => {
        console.log(isVisible);
        setVisible(!isVisible);
    };

    const dialogContentProps = {
        type: DialogType.normal,
        title: props.dialogTitle,
        subText: props.dialogText
    };

    const modalProps = {
        isDarkOverlay: true,
        className: `${props.stylingClassName}`
    };

    return (
        <>
            <Dialog hidden={!isVisible} dialogContentProps={dialogContentProps} modalProps={modalProps}>
                <DialogFooter>
                    <PrimaryButton text={i18next.t('HOME')} onClick={() => props.defaultButtonAction()} />
                    <DefaultButton text={i18next.t('CLOSE')} onClick={toggle} />
                    {/* <DefaultButton text={i18next.t('CLOSE')} /> */}
                </DialogFooter>
            </Dialog>
        </>
    );
}
