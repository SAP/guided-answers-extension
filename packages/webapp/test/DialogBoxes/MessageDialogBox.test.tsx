import React from 'react';
import i18next from 'i18next';
import { MessageDialogBox } from '../../src/webview/ui/components/DialogBoxes/MessageDialogBox';
import { render, cleanup } from '@testing-library/react';
import { initI18n } from '../../src/webview/i18n';

describe('<MessageDialogBox />', () => {
    initI18n();
    afterEach(cleanup);

    it('Should render a MessageDialogBox component', () => {
        const { container } = render(
            <MessageDialogBox
                dialogTitle={i18next.t('THANKS')}
                dialogText={i18next.t('THANK_YOU_TEXT')}
                dialogVisible={true}
                primaryButtonAction={jest.fn()}
                defaultButtonAction={jest.fn()}
                stylingClassName="solved-message-dialog"
            />
        );
        expect(container).toMatchSnapshot();
    });
});
