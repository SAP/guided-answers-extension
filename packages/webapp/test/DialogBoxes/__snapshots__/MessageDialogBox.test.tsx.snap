// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageDialogBox /> Should render a MessageDialogBox component 1`] = `
<div
  aria-hidden="true"
>
  <span
    class="ms-layer"
  />
</div>
`;
